### PR TITLE
Support react v0.14.0

### PR DESCRIPTION
**Pull request by cesarandreu**
Original PR: #15 

Moving to react-14 branch to release a prerelease.

### CODE REVIEW

**Comment by taion** [comment]
BTW, I messed up with the R-B pre-release versioning. I forgot that `v0.24.5-react-pre.0` < `v0.24.5`.

File: test/ModalSpec.js [hunk 1]
```diff
@@ -28,8 +29,7 @@ describe('Modal', function () {
       </Modal>
     , mountPoint);
 
-    assert.ok(
-      ReactTestUtils.findRenderedDOMComponentWithTag(instance.refs.modal, 'strong'));
+    assert.equal(ReactDOM.findDOMNode(instance.refs.modal).querySelectorAll('strong').length, 1);
```
Comment (taion) [inline test/ModalSpec.js]: `s/ReactDOM\.findDOMNode\(instance\.refs\.modal\)/instance.refs.modal/`

File: test/ModalSpec.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ describe('Modal', function () {
 
     let instance = render(<Container />, mountPoint);
     let modal = ReactTestUtils.findRenderedComponentWithType(instance, Modal);
-    let backdrop = React.findDOMNode(modal.refs.backdrop);
+    let backdrop = ReactDOM.findDOMNode(modal.refs.backdrop);
```
Comment (taion) [inline test/ModalSpec.js]: `s/ReactDOM\.findDOMNode\(instance\.refs\.backdrop\)/instance.refs.backdrop/`

File: test/PositionSpec.js [hunk 3]
```diff
@@ -57,7 +58,7 @@ describe('Position', function () {
               <div ref="bar" />
 
               <Position
-                target={() => React.findDOMNode(this.refs[this.state.target])}
+                target={() => ReactDOM.findDOMNode(this.refs[this.state.target])}
```
Comment (taion) [inline test/PositionSpec.js]: Drop the `ReactDOM.findDOMNode` here as well

File: src/Portal.js [hunk 4]
```diff
@@ -62,7 +63,7 @@ let Portal = React.createClass({
     // Save reference for future access.
     if (overlay !== null) {
       this._mountOverlayTarget();
-      this._overlayInstance = React.render(overlay, this._overlayTarget);
+      this._overlayInstance = ReactDOM.render(overlay, this._overlayTarget);
```
Comment (taion) [inline src/Portal.js]: Not sure if this is in `v0.14.0-beta1` but it should be `ReactDOM. unstable_renderSubtreeIntoContainer` per https://github.com/react-bootstrap/react-bootstrap/blob/v0.24.5-react-pre.0/src/Portal.js#L55

Comment (cesarandreu) [inline src/Portal.js]: I updated it. It works with v0.14.0-beta1 as well :D

File: package.json [hunk 5]
```diff
@@ -75,8 +74,8 @@
     "mt-changelog": "^0.6.1",
     "node-libs-browser": "^0.5.2",
     "raw-loader": "^0.5.1",
-    "react": "^0.13.3",
-    "react-bootstrap": "^0.24.2",
```
Comment (jquense) [inline package.json]: you should be able to use ` "react-bootstrap": "0.24.5-react-pre.0"` here its the react 014 branch

Comment (taion) [inline package.json]: Make sure you don't specify a semver range, though. I messed up the versioning (but I don't even know how to version it properly).

**Comment by jquense** [comment]
@cesarandreu  if you want to squash I'll merge and send out a release

**Comment by cesarandreu** [comment]
@jquense ready to go :D

**Comment by jquense** [comment]
ya!

**Pull request merged**